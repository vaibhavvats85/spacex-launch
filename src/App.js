import './App.scss';
import Filters from './components/filters/filters';
import Programs from './components/programs/programs';
import * as constants from './constants';
import React from 'react';

function App() {
  return (
    <div className="App">
      <div className="header">
        <Header title={constants.APP_TITLE} />
      </div>
      <div className="main">
        <Filters />
        <Programs />
      </div>
      <div className="signature">
        <Signature name={constants.DEVELOPER_NAME} />
      </div>
    </div>
  );
}

const Header = ({ title }) => (
  <span>{title}</span>
);


const Signature = ({ name }) =>
  (<>
    <strong>{constants.DEVELOPED_BY}: </strong>
    <span>{name}</span>
  </>
  );

export default App;
