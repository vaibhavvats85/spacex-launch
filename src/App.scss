@import "./shared-styles/colors";

.App {
  padding: 0 20px 20px 20px;
  .header {
    font-weight: 800;
    font-size: 40px;
  }
  .main {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-top: 20px;
  }

  .signature {
    text-align: center;
    position: fixed;
    bottom: 0;
    background-color: #f2f2f2;
    height: 34px;
    width: 100%;
  }
}

@media only screen and (max-width: 700px) {
  .App > .main {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}
