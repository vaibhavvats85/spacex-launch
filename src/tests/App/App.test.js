import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from '../../App';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<App />);
});

test('should render app component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render all child components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find('Header').exists()).toBeTruthy();
  expect(wrapper.find('Filters').exists()).toBeTruthy();
  expect(wrapper.find('Programs').exists()).toBeTruthy();
  expect(wrapper.find('Signature').exists()).toBeTruthy();
});


