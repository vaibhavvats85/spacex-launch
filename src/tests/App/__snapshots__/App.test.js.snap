// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render app component correctly 1`] = `
<div
  className="App"
>
  <div
    className="header"
  >
    <Header
      title="SpaceX Launch Programs"
    />
  </div>
  <div
    className="main"
  >
    <Filters />
    <Programs />
  </div>
  <div
    className="signature"
  >
    <Signature
      name="Vaibhav Vats"
    />
  </div>
</div>
`;
